fdk-aac: fix the license

FavoritoHJS opened the pull request:
### WARNING: NOOB AT THE WHEEL, PLEASE SCREAM ~~IF~~ WHEN I DO SOMETHING VERY WRONG

## Description of changes

Instead of using the Apache 2.0 license, add and use a custom `fraunhofer-fdk` license to prevent future confusion.

Questions: Does this license count as "free"? Am I using the correct name? Is there a viable domain link for this?

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

trofi commented on lib/licenses.nix:
```diff
@@ -481,6 +481,10 @@ in mkLicense lset) ({
   };
 
+  fraunhofer-fdk = {
+    fullName = "Software License for The Fraunhofer FDK AAC Codec Library";
```
Looks great!

I think you can add a known `spdxId`, and probably use full name from there as well:: https://spdx.org/licenses/FDK-AAC.html . Maybe something like:

```nix
spdxId = "FDK-AAC";
fullName = "Fraunhofer FDK AAC Codec Library";
```

FavoritoHJS replied:
Done, though i'm not sure if adding a new commit is the best way to do this or if i should rewrite the commit to have had the correct info in the first place...

alyssais commented:
> Is there a viable domain link for this?

What about https://sourceforge.net/p/opencore-amr/fdk-aac/ci/master/tree/NOTICE?

> Does this license count as "free"?

Looks like it, see e.g. https://www.gnu.org/licenses/license-list.en.html#fdk

Fedora seems to be changing their mind about it due to the lack of patent license, but that would also apply to e.g. CC0, which we consider to be free.

---

Please make sure your commit messages are formatted according to CONTRIBUTING.md.

FavoritoHJS committed:
licenses: add Fraunhofer FDK AAC Codec Library

This license is used by the Fraunhofer FDK codec, which is currently
mislicensed to Apache 2.0.

Of course, this can't be corrected without having the proper license
available in lib.licenses can't it??

Thanks @trofi for noticing this has an SPDX ID.

FavoritoHJS committed:
fdk-aac: change license to fraunhofer-fdk

This library was originally mislicensed as Apache 2.0, as that's the
license used by the project that has this library.

However the library still has the original license on it, so i'm fairly
certain that's the wrong license.

FavoritoHJS commented:
apparently you're meant to ignore the fact there isn't a license in the name... ok then.

reopening since this should be fixed now.

vcunat commented:
Oh I see now what you meant.  I don't know... it's weird that they don't have it, but I think it's OK to keep it the way they named it 🤷🏽
